src/views/TrackPositionAnimation/TrackPositionAnimationView.tsx

```diff
@@ -185,39 +176,34 @@ const TrackPositionAnimationView: FunctionComponent<TrackPositionAnimationProps>
             height={drawHeight}
             trackBucketRectanglesPx={trackBins}
         />, [width, drawHeight, trackBins])
-
+        
+    const configuredPositionDrawFn = useConfiguredPositionDrawFunction('plasma')
     const probabilityLayer = useMemo(() => <TPADecodedPositionLayer
             width={width}
             height={drawHeight}
             drawData={currentProbabilityFrame}
-        />, [width, drawHeight, currentProbabilityFrame])
+            configuredDrawFnCallback={configuredPositionDrawFn}
+        />, [width, drawHeight, currentProbabilityFrame, configuredPositionDrawFn])
 
     const positionLayer = useMemo(() => <TPAPositionLayer
             width={width}
             height={drawHeight}
             drawData={currentPositionFrame}
         />, [width, drawHeight, currentPositionFrame])
 
-    const controlLayer = useMemo(() => <AnimationPlaybackControls
+    return (
+        <FrameAnimation
             width={width}
-            height={controlsHeight}
-            verticalOffset={drawHeight}
+            height={height}
+            controlsHeight={controlsHeight}
+            state={animationState}
             dispatch={animationStateDispatch}
-            totalFrameCount={animationState.frameData.length}
-            currentFrameIndex={animationState.currentFrameIndex}
-            isPlaying={animationState.isPlaying}
-            playbackRate={animationState.replayMultiplier}
-        />, [width, drawHeight, animationState.frameData.length, animationState.currentFrameIndex, animationState.isPlaying, animationState.replayMultiplier])
- 
-    return (
-        <Fragment>
-            <div>
-                {trackLayer}
-                {probabilityLayer}
-                {positionLayer}
-                {controlLayer}
-            </div>
-        </Fragment>
+            dataSeriesFrameRateHz={samplingFrequencyHz}
+        >
+            {trackLayer}
```

Comment:
Here the content layers are passed as children of the `FrameAnimation` component, which is responsible for the control bars. No need for future animations to duplicate control-bar code.